pyproject.toml

```
@@ -9,9 +9,9 @@ authors = [
 ]
 description = "Tooling for building derived datasets in BigQuery"
 readme = "README.md"
-requires-python = ">=3.11"
+requires-python = ">=3.11, <3.12"
```

Does this only support min/max specifications, i.e. can't do
```suggestion
requires-python = "=3.11"
```

---

Ok that makes sense. > and < works for me

---

Unfortunately `./bqetl bootstrap` [installs `bqetl`'s dependencies](https://github.com/mozilla/bigquery-etl/blob/d213edbec6f909c2e475d6a8f567aedf2c677224/bqetl#L27) before installing `bqetl` itelf, so issue #5711 still happens even with this in place.

I'm thinking adding a Python version check to the `./bqetl` script would probably be the most user-friendly, but the downside is we'd then have to remember to maintain that Python version check logic.

---

We could have a checklist of things to do when changing the supported versions, e.g. update Dockerfile, pyproject.toml, CI, and bqetl bootstrap, but I think having any check with a clear message is an improvement